app/src/pipelines/generate_referrals/pipeline_wrapper.py

```diff
@@ -48,25 +49,54 @@ def setup(self) -> None:
         pipeline = Pipeline()
         pipeline.add_component("llm", AmazonBedrockChatGenerator(model=model))
 
-        prompt_template = haystack_utils.get_phoenix_prompt("generate_referrals")
+        # set up the pipeline with the default prompt_version_id: latest if local, configuration value if docker/deployed
+        prompt_template = haystack_utils.get_phoenix_prompt(
+            "generate_referrals"
+        )  # TODO MRH replace with component
         pipeline.add_component(
             instance=ChatPromptBuilder(
                 template=prompt_template, required_variables=["query", "supports", "resource_json"]
             ),
             name="prompt_builder",
         )
+        pipeline.add_component("load_supports", components.LoadSupports())
         pipeline.connect("prompt_builder", "llm.messages")
-
+        pipeline.connect("load_supports.supports", "prompt_builder.supports")
         self.pipeline = pipeline
 
     # Called for the `generate-referrals/run` endpoint
-    def run_api(self, query: str) -> dict:
-        supports_from_db = format_support_strings()
+    def run_api(self, query: str, prompt_version_id: str = "") -> dict:
+        if prompt_version_id:
```

Comment:
This can be simplified to much fewer changed lines: run `get_phoenix_prompt()` using the `prompt_version_id` or `config.PROMPT_VERSIONS[prompt_name]`, then run the pipeline (instead of the big `if`-block.